src/shogun/multiclass/CoverTreeKNNSolver.cpp

```diff
@@ -66,11 +66,12 @@ if (io->get_loglevel()<= MSG_DEBUG)
 		output->set_label(res[i][0].m_index, out_idx+m_min_label);
 	}
 
-
+	SG_UNREF(r);
```

Comment:
fixing missing UNREFS should go to a separate mini patch that clearly shows whats going on.